### ISSUE TITLE
[BUG] Python impl of DeserializeStage does not emit messages until receiving an on_complete from upstream

### ISSUE THREAD

**Issue by dagardner-nv** [nv-morpheus/Morpheus#299]
**Describe the bug**
This only occurs with the Python impl of the stage. I suspect a bug in `srf.core.ops.flatten` but I am not sure.

**Steps/Code to reproduce bug**
```python
import logging
import os
import time

import srf
import cudf

from morpheus.config import Config
from morpheus.config import CppConfig
from morpheus.config import PipelineModes
from morpheus.messages import MessageMeta
from morpheus.pipeline.linear_pipeline import LinearPipeline
from morpheus.pipeline.single_output_source import SingleOutputSource
from morpheus.pipeline.stream_pair import StreamPair
from morpheus.stages.general.monitor_stage import MonitorStage
from morpheus.stages.output.write_to_file_stage import WriteToFileStage
from morpheus.stages.postprocess.serialize_stage import SerializeStage
from morpheus.stages.preprocess.deserialize_stage import DeserializeStage
from morpheus.utils.logging import configure_logging

class SimpleSrc(SingleOutputSource):
    def __init__(self, c: Config, count=10) -> None:
        super().__init__(c)
        self.count = count

    @property
    def name(self) -> str:
        return "simple"

    def supports_cpp_node(self):
        return False

    def _generate_frames(self):
        for i in range(self.count):
            df = cudf.DataFrame({"col": [i]})
            m = MessageMeta(df)

            if i == self.count - 1:
                # Sleep for 20 seconds before emitting the last message
                time.sleep(20)

            yield m

    def _build_source(self, builder: srf.Builder) -> StreamPair:
        src = builder.make_source(self.unique_name, self._generate_frames)
        return src, MessageMeta

def run_pipeline():
    configure_logging(log_level=logging.DEBUG)
    config = Config()
    CppConfig.set_should_use_cpp(False)
    config.mode = PipelineModes.NLP
    config.num_threads = os.cpu_count()
    config.feature_length = 256

    print("Config: \n%s", config.to_string())
    print("CPP Enabled: {}".format(CppConfig.get_should_use_cpp()))

    pipeline = LinearPipeline(config)
    pipeline.set_source(SimpleSrc(config))
    pipeline.add_stage(MonitorStage(config, description="Source rate"))
    pipeline.add_stage(DeserializeStage(config))
    pipeline.add_stage(MonitorStage(config, description="Deserialize rate"))
    pipeline.add_stage(SerializeStage(config))
    pipeline.add_stage(MonitorStage(config, description="Serialize rate"))
    pipeline.add_stage(WriteToFileStage(config, filename="/tmp/test-source.csv", overwrite=True))
    pipeline.run()

if __name__ == "__main__":
    run_pipeline()
```

If I run that script, the monitor stage does not log any messages being emitted from the `DeserializeStage` until after the 20 second sleep that the source performs.

If however I enable C++ execution, then the monitor stages for `DeserializeStage` & `WriteToFileStage` log messages as they are being generated by the source.

**Expected behavior**
Emit messages as they are received

**Additional context**
Spun out of https://github.com/nv-morpheus/Morpheus/issues/297

**Comment by dagardner-nv** [comment]
Caused by https://github.com/nv-morpheus/SRF/issues/116

**Comment by dagardner-nv** [comment]
Fixed in srf-22.06.01
